week7/cooper_client_react/02_add_login.md

```diff
@@ -0,0 +1,371 @@
+  })
+})
+
+```
+
+This feature is pretty straight forward. We click a login button which renders a login form. In the first scenario, we fill in the correct credentials and in the second one we fill in the wrong ones. Depending on if it is successful or not we will get a response which either welcomes the user or returns an error message.
+
+If we run the test now we get an error that states that the test can't find the selector `#login`. So let's start with adding that button to our App component.
+
+```js
+// src/App.js
+
+...
+
+ render() {
+    return (
+      <div>
+        <InputFields 
+          inputChangeHandler={this.onChange.bind(this)}
+        />
+
+        <button id="login">Login</button>
+
+        <DisplayCooperResult
+          distance={this.state.distance}
+          gender={this.state.gender}
+          age={this.state.age}
+        />
+      </div>
+    );
+  }
+}
+```
+So that takes care of that error. Now we have a different error that is very similiar to the previous one. It cant find the input field for filling in the email.
+At this point, we want to create a new component for the login form. This component should have input fields for the credentials and a button to submit them.
+
+Let's create this login form component:
+
+`$ mkdir src/Components`
+`$ touch src/Components/LoginForm.js`
+
+Add this to that file:
+
+```js
+import React from 'react';
+
+const LoginForm = () => {
+  return (
+    <form>
+      <div>
+        <label >Email</label>
+        <input id="email"></input>
+      </div>
+
+      <div>
+        <label>Password</label>
+        <input id="password"></input>
+      </div>
+      <button id="submit">Submit</button>
+    </form>
+  )
+}
+
+export default LoginForm;
+
+```
+
+If we run the test again, we will see no difference in the output. That is because we don't use this component anywhere, we don't render it.
+In the app component, we need to import it and render it.
+
+Modify `App.js` with the following code.
+
+```js
+import React, { Component } from 'react';
+import './App.css';
+import DisplayCooperResult from './DisplayCooperResult'
+import LoginForm from './Components/LoginForm';
+
+
+class App extends Component {
+ // ...
+
+ render() {
+    return (
+      <div>
+        // ...
+
+        <button id="login">Login</button>
+        <LoginForm />
+      </div>
+    );
+  }
+}
+
+export default App;
+```
+
+If you run the feature test now, it will complain about not finding the text that we are expecting to find.
+So at the moment, the test can find all of the elements it needs to fulfill its task. The flow that we want is that when the user clicks the login button we want the login form to render, at the moment we have a button that does nothing. If you run the application you can see that.
+
+To fix this you need to update the App component to look like this:
+```js
+import React, { Component } from 'react';
+import './App.css';
+import DisplayCooperResult from './DisplayCooperResult';
```

Comment:
wrong path, should be `import DisplayCooperResult from './Components/DisplayCooperResult';`